proto/regen/ecocredit/basket/v1alpha1/tx.proto

```diff
@@ -0,0 +1,118 @@
+syntax = "proto3";
+
+package regen.ecocredit.basket.v1alpha1;
+
+option go_package = "github.com/regen-network/regen-ledger/x/ecocredit/basket/v1alpha1";
+
+import "google/protobuf/duration.proto";
+import "regen/ecocredit/basket/v1alpha1/types.proto";
+
+// Msg is the regen.ecocredit.basket.v1beta1 Msg service.
+service Msg {
+
+  // CreateBasket creates a bank denom which wraps credits.
+  rpc CreateBasket(MsgCreateBasket) returns (MsgCreateBasketResponse);
+
+  // AddToBasket adds credits to a basket in return for basket tokens.
+  rpc AddToBasket(MsgAddToBasket) returns (MsgAddToBasketResponse);
+
+  // TakeFromBasket takes credits from a basket starting from the oldest
+  // credits first.
+  rpc TakeFromBasket(MsgTakeFromBasket) returns (MsgTakeFromBasketResponse);
+}
+
+// MsgCreateBasket is the Msg/CreateBasket request type.
+message MsgCreateBasket {
+  // curator is the address of the basket curator who is able to change certain
+  // basket settings.
+  string curator = 1;
+
+  // name will be used to create a bank denom for this basket token of the form
+  // ecocredit:{curator}:{name}.
```

Comment:
```suggestion
  // name will be used to create a bank denom for this basket token
```
omit, `:` isnt valid in denoms

Reply:
I had a PR in the SDK to change the regex. Did it not get merged?

Reply:
perhaps is not in our fork? just tested on `github.com/regen-network/cosmos-sdk v0.44.2-regen-1` and it fails